versions/development/SPEC.md

```diff
@@ -689,6 +691,59 @@ Pair values can also be specified within the [workflow inputs JSON](https://gith
 }
 ```
 
+### Optional literals
+
+Any non-optional value can be stored into an optional variable by assigning it to an optional variable.
+This does not change the value, but it changes the type of the value:
+
+```WDL
+Int? maybe_five = 5
+```
+
+The `None` literal is the value that an optional has when it is not defined.
+
+```WDL
+Int? maybe_five_but_is_not = None
+# maybe_five_but_is_not is an undefined optional
+Int? maybe_five_and_is = 5
+# maybe_five_and_is is a defined optional
+Int certainly_five = 5
+# Certainly five is not an optional
+
+Boolean test_defined = defined(maybe_five_but_is_not) # Evaluates to false
+Boolean test_defined2 = defined(maybe_five_and_is) # Evaluates to true
+Boolean test_is_none = maybe_five_but_is_not == None # Evaluates to true, same as !definedmaybe_five_but_is_not)
+Boolean test_not_none = maybe_five_but_is_not != None # Evaluates to false, same as defined(maybe_five_but_is_not)
+Boolean compare_int_to_none = certainly_five == None # This will cause an error, since None only makes sense when dealing with optionals.
+```
+
+This is useful in a number of cases, for example when an output of a command depends on a certain flag:
+```wdl
+task test {
+  input {
+    File inputFile
+    String outputPath
+    Boolean log
+  }
+  command {
+    test ~{true="--log" false="" log} inputFile outputPath
+  }
+  output {
+    File? logFile = if log then "test.log" else None
```

Comment:
This is the line which I think could be trickier to guarantee. I think for this PR that this example should be removed or reworked to not use `File?`, and a new PR specifically guaranteeing support for `File?` outputs should be added (in case there are alternative ways or syntax options of expressing this intention in an easier-for-cloud-support way)